pennylane/fourier/visualize.py

```diff
@@ -0,0 +1,720 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains visualization functions for Fourier series and coefficients."""
+import warnings
+from itertools import product
+import numpy as np
+
+# pylint:disable=too-many-arguments,blacklisted-name
+
+# Matplotlib is not a hard requirement for PennyLane in general, but it *is*
+# a hard requirement for everything in this module.
+try:
+    from matplotlib.colors import to_rgb
+except ModuleNotFoundError:
+    warnings.warn("Module matplotlib is required for visualization in the Fourier module.")
```

Comment:
If it is a hard requirement, then should an error be raised here?

Reply:
In this case couldn't we do that by just leaving `import matplotlib.pyplot as plt` without anything else around it?

Reply:
Yep, although it would still be nice to do

```python
try:
    from matplotlib.colors import to_rgb
except (ModuleNotFoundError, ImportError) as e:
    raise ImportError(
        "Matplotlib is required for visualization in the Fourier module. "
        "You can install matplotlib via \n\n   pip install matplotlib"
    ) from e
```

as it provides a less scary import error lol

Reply:
Haha fair enough, this is much more friendly :smile:

Reply:
Changed it, good point